blue_modules/currency.ts

```
@@ -27,11 +29,111 @@
+    const rate = await getFiatRate(preferredFiatCurrency.endPointKey);
+    exchangeRates[LAST_UPDATED] = Date.now();
+    exchangeRates[BTC_PREFIX + preferredFiatCurrency.endPointKey] = rate;
+    exchangeRates.LAST_UPDATED_ERROR = false;
+    await AsyncStorage.setItem(EXCHANGE_RATES_STORAGE_KEY, JSON.stringify(exchangeRates));
+  } catch (error: any) {
+    console.error('Error encountered when attempting to update exchange rate...', error);
+
+    // Check for specific server error codes
+    if (error.response?.status && [500, 502, 503, 504].includes(error.response.status)) {
+      if (!suppressExchangeRateAlert) {
+        await presentCurrencyFetchOrParseError(preferredFiatCurrency.endPointKey);
```

Biome

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)